src/main/java/io/confluent/connect/hdfs/FileUtils.java

```diff
@@ -151,7 +152,12 @@ public static FileStatus fileStatusWithMaxOffset(Storage storage, Path path,
   }
 
   public static long extractOffset(String filename) {
-    return Long.parseLong(filename.split(HdfsSinkConnecorConstants.COMMMITTED_FILENAME_SEPARATOR_REGEX)[3]);
+    Matcher m = HdfsSinkConnectorConstants.COMMITTED_FILENAME_PATTERN.matcher(filename);
+    // NB: if statement has side effect of enabling group() call
+    if (!m.matches()) {
+      throw new IllegalArgumentException(filename + " does not match COMMITTED_FILENAME_PATTERN");
+    }
+    return Long.parseLong(m.group(4));
```

Comment:
Should we maybe change this to a named constant in `HdfsSinkConnectorConstants` since it could change if, e.g. any grouping was adjusted in `COMMITTED_FILENAME_PATTERN`?